cogs/boolean_utils.py

```diff
@@ -0,0 +1,395 @@
+"""
+discord bot cog that is used for various number utility functions
+"""
+import discord
+from discord.ext import commands
+
+class BoolUtilsCog:
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def bool(self, ctx, exp):
+        """Creates truth table from expression using !,*,^,+
+
+        Wrap expression in "quotes" to include spaces in your expression.
+        If there are spaces in your expression and it is not properly wrapped,
+        then it will not process correctly, if at all.
+
+        Single chars are variables, more than one char in a row is invalid.
+        More than one ! in a row is invalid... fight me.
+        More than 5 variables is invalid, otherwise the messages would be long.
+        """
+        await ctx.send(get_bool_table(exp))
+
+def setup(bot):
+    """Sets up the cog"""
+    bot.add_cog(BoolUtilsCog(bot))
+
+def get_bool_table(exp):
+    """Parses a boolean expression and creates a truth table"""
+    boolean_exp = BooleanExpr(exp)
+    return "```" + boolean_exp.get_truth_table() + "```"
+
+#####################################################################
+#####################################################################
+#### Class to parse and evaluate boolean expressions ################
+#####################################################################
+#####################################################################
+
+class BooleanExpr:
+    # Expression with whitespace stripped
+    orig_exp = "empty"
+    # How the expression will be displayed
+    disp_exp = "empty"
+    # Post-fix version of the original expression
+    post_exp = ""
+    # Set of all the variables
+    var_set = set()
+    # Stack needed to verify correctness of expr
+    var_stack = []
+    # Stack to manage operator precedence
+    op_stack = []
+    # The formatted truth table to be sent on discord
+    result_formatted = ""
+    # Flag and message in case an error occurs
+    error = False
+    error_msg = ""
+
+    # Defines operator precedence
+    prec_dict = {
+        '!' : 0, # NOT
+        '*' : 1, # AND
```

Comment:
Does this interpret `AB` as `A*B`?

Reply:
No. I don't think it would be worth the effort to implement that behavior.